src/NodeParser/TypeReferenceNodeParser.ts

```diff
@@ -53,6 +47,16 @@ export class TypeReferenceNodeParser implements SubNodeParser {
             return context.getArgument(typeSymbol.name);
         }
 
+        // Wraps promise type to avoid resolving to a empty Object type.
+        if (typeSymbol.name === "Promise" || typeSymbol.name === "PromiseLike") {
+            // Promise without type resolves to Promise<any>
+            if (!node.typeArguments || node.typeArguments.length === 0) {
+                return new AnyType();
```

Comment:
Can we move this logic into the PromiseNodeParser?

Reply:
Actually not, It might be a skill issue of mine but I couldn't get it to work without this statement.

When I remove it, this is the error thrown:

```console
  ● valid-data-type › promise-extensions

    TypeError: Cannot read properties of undefined (reading 'getId')

      123 |
      124 |     // Check for simple type equality
    > 125 |     if (source.getId() === target.getId()) {
          |                ^
      126 |         return true;
      127 |     }
      128 |

      at getId (src/Utils/isAssignableTo.ts:125:16)
      at src/NodeParser/ConditionalTypeNodeParser.ts:44:77
      at predicate (src/Utils/narrowType.ts:59:12)
      at ConditionalTypeNodeParser.createType (src/NodeParser/ConditionalTypeNodeParser.ts:44:41)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:35:54)
      at TypeAliasNodeParser.createType (src/NodeParser/TypeAliasNodeParser.ts:40:43)
      at AnnotatedNodeParser.createType (src/NodeParser/AnnotatedNodeParser.ts:34:47)
      at ExposeNodeParser.createType (src/ExposeNodeParser.ts:23:45)
      at CircularReferenceNodeParser.createType (src/CircularReferenceNodeParser.ts:24:43)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:35:54)
      at TypeReferenceNodeParser.createType (src/NodeParser/TypeReferenceNodeParser.ts:78:37)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:35:54)
      at PromiseNodeParser.createType (src/NodeParser/PromiseNodeParser.ts:71:47)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:35:54)
      at TypeReferenceNodeParser.createType (src/NodeParser/TypeReferenceNodeParser.ts:78:37)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:35:54)
      at createType (src/NodeParser/UnionNodeParser.ts:22:45)
          at Array.map (<anonymous>)
      at UnionNodeParser.map [as createType] (src/NodeParser/UnionNodeParser.ts:21:14)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:35:54)
      at TypeAliasNodeParser.createType (src/NodeParser/TypeAliasNodeParser.ts:40:43)
      at AnnotatedNodeParser.createType (src/NodeParser/AnnotatedNodeParser.ts:34:47)
      at ExposeNodeParser.createType (src/ExposeNodeParser.ts:23:45)
      at CircularReferenceNodeParser.createType (src/CircularReferenceNodeParser.ts:24:43)
      at ChainNodeParser.createType (src/ChainNodeParser.ts:35:54)
      at TopRefNodeParser.createType (src/TopRefNodeParser.ts:14:47)
      at createType (src/SchemaGenerator.ts:30:36)
          at Array.map (<anonymous>)
      at SchemaGenerator.map [as createSchemaFromNodes] (src/SchemaGenerator.ts:29:37)
      at SchemaGenerator.createSchemaFromNodes [as createSchema] (src/SchemaGenerator.ts:25:21)
      at Object.createSchema (test/utils.ts:61:34)
```